File: in_Java/0746 Min Cost Climbing Stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] dp = new int[cost.length+1];
        dp[1] = cost[0];
        int i;
        for(i = 2; i < dp.length-1; i++){
            dp[i] = Math.min(dp[i-1],dp[i-2]) + cost[i-1];
        }
        return Math.min(dp[dp.length-2],dp[dp.length-3] + cost[cost.length-1]);
    }
}

File: in_Python/0746 Min Cost Climbing Stairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        if len(cost) == 2:
            return min(cost[0], cost[1])
        dp = []
        dp.append(cost[0])
        dp.append(cost[1])
        for i in range(2, len(cost)-1):
            dp.append(min(dp[-1] + cost[i], dp[-2] + cost[i]))
        return min(dp[-1], dp[-2] + cost[-1])
        